src/components/organisms/topbar/dashboard.tsx

```diff
@@ -39,23 +52,141 @@ export const DashboardTopbar = () => {
+		const file = event.target.files?.[0];
+		if (!file) return;
+		setLoadingImportFile(true);
+
+		try {
+			const zip = await JSZip.loadAsync(file);
+			const fileStructure = await processZipContent(zip);
+
+			const manifestFileNode = fileStructure["autokitteh.yaml"];
+			const manifestContent = manifestFileNode && "content" in manifestFileNode ? manifestFileNode.content : null;
+			if (!manifestContent) {
+				addToast({
+					message: tTemplates("projectCreationFailed"),
+					type: "error",
+				});
+
+				LoggerService.error(
+					namespaces.manifestService,
+					`${tTemplates("projectCreationFailedExtended", { error: tTemplates("projectTemplateManifestNotFound") })}`
+				);
+
+				return;
+			}
+
+			const manifest = yaml.load(manifestContent);
+			const updatedManifestContent = yaml.dump(manifest);
+
+			const { data: newProjectId, error } = await createProjectFromManifest(updatedManifestContent);
+			if (error || !newProjectId) {
+				addToast({
+					message: tTemplates("projectCreationFailed"),
+					type: "error",
+				});
+				LoggerService.error(
+					namespaces.manifestService,
+					`${tTemplates("projectCreationFailedExtended", { error })}`
+				);
+
+				return;
+			}
+
+			delete fileStructure["autokitteh.yaml"];
+			delete fileStructure["autokitteh.yaml.user"];
+
+			setProjectId(newProjectId);
+			setTemplateFiles(fileStructure);
+		} catch (error) {
+			addToast({
+				message: tTemplates("projectCreationFailed"),
+				type: "error",
+			});
+
+			LoggerService.error(
+				namespaces.manifestService,
+				`${tTemplates("projectCreationFailedExtended", { error })}`
+			);
+		} finally {
+			setLoadingImportFile(false);
+			if (fileInputRef.current) {
+				fileInputRef.current.value = "";
+			}
+		}
+	};
+
+	const triggerFileInput = () => {
+		fileInputRef.current?.click();
+	};
+
 	return (
 		<div className="z-10 flex flex-wrap">
 			<div className="flex w-full items-end justify-between">
 				<Typography className="w-full font-averta text-4xl font-semibold" element="h1">
 					{t("welcome")}
 				</Typography>
 
-				<Button
-					className="gap-2.5 whitespace-nowrap rounded-full border border-gray-750 py-2.5 pl-3 pr-4 font-averta text-base font-semibold"
-					disabled={loadingNewProject}
-					onClick={handleCreateProject}
-					variant="filled"
-				>
-					<IconSvg className="fill-white" size="lg" src={!loadingNewProject ? PlusAccordionIcon : Spinner} />
-
-					{t("buttons.newProject")}
-				</Button>
+				<div className="flex rounded-full border border-gray-750 font-averta text-base font-semibold">
+					<Button
+						className="gap-2.5 whitespace-nowrap rounded-none rounded-l-full border-r border-gray-750 px-3 py-2.5"
+						disabled={loadingNewProject}
+						onClick={handleCreateProject}
+						variant="filled"
+					>
+						<IconSvg
+							className="fill-white"
+							size="lg"
+							src={!loadingNewProject ? PlusAccordionIcon : Spinner}
+						/>
```

Comment:
According the design, please remove the icon.

Reply:
removed